src/main/scala/io/iohk/ethereum/network/p2p/messages/PV63.scala

```diff
@@ -167,10 +167,15 @@ object PV63 {
   object ReceiptImplicits {
     import TxLogEntryImplicits._
 
-    implicit class ReceiptEnc(msg: Receipt) extends RLPSerializable {
+    implicit class ReceiptEnc(receipt: Receipt) extends RLPSerializable {
       override def toRLPEncodable: RLPEncodeable = {
-        import msg._
-        RLPList(postTransactionStateHash, cumulativeGasUsed, logsBloomFilter, RLPList(logs.map(_.toRLPEncodable): _*))
+        import receipt._
+        val stateHash = postTransactionStateHash match {
```

Comment:
here you can add type to stateHash like : `val stateHash: RLPEncodeable`, and then in cases you do not need to serialize it by hand i.e you can just write:

```
case HashOutcome(hash) => hash
case SuccessOutcome => 1.toByte      
case _ => 0.toByte
```
And proper serialization implicitis will kick in.